src/firefly/java/edu/caltech/ipac/util/FileUtil.java

```diff
@@ -1013,10 +1012,107 @@ private static URL getThisClassURL() {
     }
 
     private static String getThisClassFileName() {
-        String cName= FileUtil.class.getName();
+        String cName = FileUtil.class.getName();
         return cName.replace(".", "/") + ".class";
     }
 
+    /**
+     * Determines the MIME type of the given file.
+     * This method may return an incorrect MIME type if the file extension is registered,
+     * and no other detection method is available.
+     *
+     * @param file input file to detect
+     * @return A String representing the MIME type of the file, or "unknown" otherwise
+     */
+    public static String getMimeType(String file) {
+        return new Tika().detect(file);
+    }
+
+    /**
+     * Detects the format of the given file
+     *
+     * @param fname The full path of the file to detect the format for.
+     * @return The detected {@code Format} of the file, or {@code UNKNOWN} if the format could not be determined.
+     * @throws IOException If an I/O error occurs while accessing the file.
+     */
+    @Nonnull
+    public static Format detectFormat(String fname) throws IOException {
+        StopWatch.getInstance().start("detectFormat");
+        String mime = getMimeType(fname);
+        DuckDbAdapter.MimeDesc mimeDesc = null;
+        Format format = mapToFormat(mime, null);
+        if (format == null) {
```

Comment:
It appears that if it can get the format from the file extension then it does not look inside the file like it use to. I think that is dangerous. I think our own server code somethings save stuff with the wrong extension. Because code that was intended  for a specific purpose got generalized.

Reply:
Tika first checks the magic number and then the file extension. DuckDB's extension, similar to the file command, relies solely on the magic number but can reliably identify Parquet files.

The logic here is to use these two tools to determine the file format. If they fail, we fall back to trial and error based on the resolved MIME type. For example, if the file is identified as XML, we can attempt to process it using a VOTable reader.  I’ve implemented handling for common cases and expect to add more as needed.

This approach works well in most cases. However, it fails when a file lacks a magic number but has a misleading file extension—for instance, a .fits file that is actually an IPAC table.

Do we need to handle this edge case?

Reply:
> This approach works well in most cases. However, it fails when a file lacks a magic number but has a misleading file extension—for instance, a .fits file that is actually an IPAC table.

This is exactly my concern. I think there are cases that we download a file with the fits extension even it is an ipac table. I don't remember right now where. I know many of our files have a `.ul` extension. 

Before you do anything let me look into this case and see if I can find it.

Reply:
Of course.  If we need to handle misleading file extensions, I can explore the best approach. The `.ul` extension is not an issue since it's not a recognized file type.